auth/src/main/java/com/firebase/ui/auth/util/SmartlockUtil.java

```diff
@@ -12,44 +12,54 @@
  * Helper class to deal with Smartlock Flows.
  */
 public class SmartlockUtil {
-
     /**
      * If SmartLock is enabled and Google Play Services is available, start the save credential
      * Activity. Otherwise, finish the calling Activity with RESULT_OK.
-     * @param activity the calling Activity.
-     * @param requestCode request code to use when starting the save operation.
-     * @param parameters calling Activity flow parameters.
+     *
+     * @param activity     the calling Activity.
+     * @param requestCode  request code to use when starting the save operation.
+     * @param data         An Intent to be merged. Must be a {@code new Intent().putExtras(...)}.
```

Comment:
See pics:

The two differeent ways of merging intents:
![image](https://cloud.githubusercontent.com/assets/9490724/19059783/723161ec-8997-11e6-81c1-74badea899bc.png)

The left intent merge up top produces the left version below and vice versa:
![image](https://cloud.githubusercontent.com/assets/9490724/19059862/08a4006c-8998-11e6-9e98-ac722fe18cf2.png)

So basically, the left version works in terms of keeping the email alive, but the it doesn't in the sense that it says it comes from the class that it originated from eg. `SignInActivity` instead of `SaveCredentialsActivity` (the test expects `SaveCredentialsActivity`).